.github/workflows/linter.yml

```
@@ -0,0 +1,27 @@
+name: Lint Code Base
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: main
+permissions: read-all
+jobs:
+  build:
+    name: Lint Code Base
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          # Full git history is needed to get a proper
+          # list of changed files within `super-linter`
+          fetch-depth: 0
+      - name: Lint Code Base
+        uses: super-linter/super-linter@v6.3.0
```

Let's go with v6.3.1 as v6.3.0 has a bug that prevents it from running on my system and I want to be consistent with the version used in CI:
```suggestion
        uses: super-linter/super-linter@v6.3.1
```